### ISSUE TITLE
BadRequestError for vport.Protocols.add()

### ISSUE THREAD

**Issue by skayres** [OpenIxia/ixnetwork_restpy#10]
Hi! 

I'm trying to change SendP2PHellosToUnicastMac and I take this example:
```
from ixnetwork_restpy.testplatform.testplatform import TestPlatform


testplatform = TestPlatform('****')
sessions = testplatform.Sessions.add(ApplicationType='ixnrest', Name=None)
ixnetwork = sessions.Ixnetwork
vport = ixnetwork.Vport.add(ConnectedTo=None, IsPullOnly=None, Name=None, RxMode=None, TraceEnabled=None, TraceLevel=None, TraceTag=None, TransmitIgnoreLinkStatus=None, TxGapControlMode=None, TxMode=None, Type=None, UseGlobalSettings=None)
protocols = vport.Protocols.add()
isis = protocols.Isis
testplatform.info(isis.SendP2PHellosToUnicastMac)
```
In running I got the error:
```
skayres@hfts4:~/dev_autotest/test_automation$ python p2p.py
2019-09-26 13:40:59 [ixnetwork_restpy.connection] [INFO] using python version 2.7.13 (default, Sep 26 2018, 18:42:22)
[GCC 6.3.0 20170516]
2019-09-26 13:40:59 [ixnetwork_restpy.connection] [INFO] using ixnetwork-restpy version 1.0.41
2019-09-26 13:40:59 [ixnetwork_restpy.connection] [INFO] The package will automatically determine the test tool platform and rest_port using the ****** address...
2019-09-26 13:40:59 [ixnetwork_restpy.connection] [INFO] Unable to connect to test tool at http://****:443.
2019-09-26 13:40:59 [ixnetwork_restpy.connection] [INFO] Unable to connect to test tool at https://****:443.
2019-09-26 13:40:59 [ixnetwork_restpy.connection] [INFO] Connection established to `http://****:11009 on windows`
Traceback (most recent call last):
  File "p2p.py", line 8, in <module>
    protocols = vport.Protocols.add()
  File "/usr/local/lib/python2.7/dist-packages/ixnetwork_restpy/testplatform/sessions/ixnetwork/vport/protocols/protocols.py", line 408, in add
    return self._create(locals())
  File "/usr/local/lib/python2.7/dist-packages/ixnetwork_restpy/base.py", line 233, in _create
    properties = self._connection._create(url, payload)
  File "/usr/local/lib/python2.7/dist-packages/ixnetwork_restpy/connection.py", line 188, in _create
    return self._send_recv('POST', url, payload)
  File "/usr/local/lib/python2.7/dist-packages/ixnetwork_restpy/connection.py", line 376, in _send_recv
    self._process_response_status_code(response)
  File "/usr/local/lib/python2.7/dist-packages/ixnetwork_restpy/connection.py", line 280, in _process_response_status_code
    raise BadRequestError(response)
ixnetwork_restpy.errors.BadRequestError: https://****:11009/api/v1/sessions/1/ixnetwork/vport/2/protocols => 400 Bad Request {"method":"POST","pathAndQuery":"/api/v1/sessions/1/ixnetwork/vport/2/protocols","errors":[{"code":10000,"detail":"\u0421\u0441\u044b\u043b\u043a\u0430 \u043d\u0430 \u043e\u0431\u044a\u0435\u043a\u0442 \u043d\u0435 \u0443\u043a\u0430\u0437\u044b\u0432\u0430\u0435\u0442 \u043d\u0430 \u044d\u043a\u0437\u0435\u043c\u043f\u043b\u044f\u0440 \u043e\u0431\u044a\u0435\u043a\u0442\u0430."}]}
```
```
skayres@hfts4:~/dev_autotest/test_automation$ pip list | grep ixnetwork
ixnetwork-restpy              1.0.41
```

What I'm wrong?

Thank you!

**Comment by ajbalogh** [comment]
There is a bug in the server rest api OPTIONS command where the /vport/protocols node internal multiplicity is being identified as a list node instead of a required node.

To work around this issue do not do Protocols.add() but instead do a **Protocols.find()**.

This will be addressed in the next client release.

**Comment by skayres** [comment]
There is also no action regards to SendP2PHellosToUnicastMac.
I have this code:
```

from ixnetwork_restpy.testplatform.testplatform import TestPlatform
from ixnetwork_restpy.assistants.statistics.statviewassistant import StatViewAssistant

# Configure ixnetwork
..

vport = ixnetwork.Vport.add(Name='R-1 port-1')

protocols = vport.Protocols.find()
isis = protocols.Isis
print(isis.SendP2PHellosToUnicastMac)
isis.SendP2PHellosToUnicastMac = False
isis.update(SendP2PHellosToUnicastMac=False)
print(isis.SendP2PHellosToUnicastMac)

#Add a chassis and ports
..

#add abstract ports and connect them to chassis ports
..

#configure topology
..
# Configuring Ethernet
..

# Configuring IPv4
..

# Configuring ISIS
isis = ethernet.IsisL3.add()
router_isis = device_group.IsisL3Router.find()
isis.LevelType.Single('level2')
isis.NetworkType.Single('pointpoint')
router_isis.EnableWideMetric.Single(True)

# Configuring Network group
..

ixnetwork.StartAllProtocols()
```
Console output about SendP2PHellosToUnicastMac is following:
```
True
False
```

But on Ixia Scenario I have 'SendP2PHellosToUnicastMac' is checked (True).
![send_p2p_unicast_enabled](https://user-images.githubusercontent.com/10278524/65721092-10131f80-e0b2-11e9-9973-fc3d8fe3cae1.PNG)


But on Ixia API Browser I have 'SendP2PHellosToUnicastMac' is false (uncheked).
![send_p2p_unicast_disabled](https://user-images.githubusercontent.com/10278524/65721100-16090080-e0b2-11e9-93b2-679b73fcfd56.PNG)


Realy Ixia sends isis p2p hello to Unicast:
![send_p2p_unicast_disabled_but_not](https://user-images.githubusercontent.com/10278524/65721115-1bfee180-e0b2-11e9-9503-df3ac572c1e5.PNG)

**Comment by ajbalogh** [comment]
Configuring classic protocols does not affect the configuration of NGPF protocols.

**The following configures classic protocols:
```
protocols = vport.Protocols.find()
isis = protocols.Isis
print(isis.SendP2PHellosToUnicastMac)
isis.SendP2PHellosToUnicastMac = False
isis.update(SendP2PHellosToUnicastMac=False)
print(isis.SendP2PHellosToUnicastMac)
```

**The following configures NGPF (next generation protocol framework) protocols
```
isis = ethernet.IsisL3.add()
router_isis = device_group.IsisL3Router.find()
isis.LevelType.Single('level2')
isis.NetworkType.Single('pointpoint')
router_isis.EnableWideMetric.Single(True)
```

**Comment by ajbalogh** [comment]
```
isis = ethernet.IsisL3.add()
isis.LevelType.Single('level2')
isis.NetworkType.Single('pointpoint')
isis.SuppressHello.Single(True)
```

**Comment by skayres** [comment]
This attribute:
`isis.SuppressHello.Single(True)`
just turn off any hello isis not change the destination MAC address from Unicast to Multicast.

SendP2PHellosToUnicastMac is still enabled:
![send_p2p_unicast_enabled_2](https://user-images.githubusercontent.com/10278524/65752137-d4f60800-e114-11e9-8437-ce2f00ffc681.PNG)

And there is no any hello from Ixia to DUT:
![send_p2p_unicast_disabled_supress](https://user-images.githubusercontent.com/10278524/65752187-eb03c880-e114-11e9-9256-997ce6d9b0bf.PNG)
(It's a control-plane capture. Hellos to Multicast p2p MAC are from DUT only)

Is there any way to change the destination MAC address for p2p isis hello form NGPF?

**Comment by ajbalogh** [comment]
```
from ixnetwork_restpy.testplatform.testplatform import TestPlatform


testplatform = TestPlatform('127.0.0.1')
sessions = testplatform.Sessions.add(ApplicationType='ixnrest', Name=None)
ixnetwork = sessions.Ixnetwork
globals = ixnetwork.Globals
topology = globals.Topology
isisl3router = topology.IsisL3Router.find(Count=None, DescriptiveName=None, Name=None, RowNames=None)
sendp2phellostounicastmac = isisl3router.SendP2PHellosToUnicastMAC
sendp2phellostounicastmac.Single(False)
```

**Comment by skayres** [comment]
Yes, it works.
Thank you very much!

**Comment by ajbalogh** [comment]
fixed vport.protocols multiplicity so that only find gets generated
available in 1.0.44
